### PR TITLE
Short term improvements to the execution plan view

### PR DESCRIPTION
**Pull request by bengotow**
This PR introduces a new "display event" type client-side that can be used to embed additional key-value data in the execution plan view. The file `RunMetadataProvider` controls the mapping from GraphQL log events to instances of this type attached to steps. Right now, the materialization event is the only one that generates one of these display events.

Display events can be "hidden", in which case they appear within a disclosure triangle shown in the step:

image

image

You can also set hidden: false to display data beneath the box all of the time (the old approach). It wasn't a lot of work to keep both visual styles intact, and I figure we can play with the RunMetadataProvider mapping to see which UI makes sense for various kinds of event info.

image

image

There are three different display modes that these items appear in:
- If the value is a path, the format is `key [copy path]`.
- if the value is a URL, the format is `key [open link]`.
- for any other value, the format is `key: value`, where clicking the value copies it.

We could also just show the paths / links, but just playing with the dagstermill materializations this seemed like it'd clutter the UI with (mostly) long and arbitrary strings.

Display events also have a "type" field which is not used at this time, but I figured we could add some iconography based on this once we have some more display events implemented.

### CODE REVIEW

**Comment by schrockn** [comment]
This is looking cool. One other thing I would like to be able to do is to have the ability to render a more complicated object (e.g. a great expectations metadata result) but essentially have the generic capability of having the user attach a bag of key-value pairs onto any of these things and then have it display sensibly.

**Comment by bengotow** [comment]
Updated this to be less smart and more straightfoward to customize. In addition to the open link and copy value actions, there's a new "show modal" option that presents the entire value in a code-formatted dialog:

image

DisplayEvent schema:

```


export interface IStepDisplayEvent {
  key: string;
  value: string;
  summary?: string;
  hidden: boolean; // true to place inside disclosure triangle
  icon:
    | "dot-success"
    | "dot-failure"
    | "dot-pending"
    | "file"
    | "link"
    | "none";
  action: "open-in-tab" | "copy" | "show-in-modal" | "none";
}
```

Here's what generated the output above:
```
          step.displayEvents.push({
            icon: "dot-success",
            key: "expect_plan_upload_success",
            summary: "click to view",
            value:
              "https://www.nytimes.com/2019/05/08/us/politics/donald-trump-jr-senate-subpoena.html?action=click&module=Top%20Stories&pgtype=Homepage",
            hidden: true,
            action: "open-in-tab"
          });
          step.displayEvents.push({
            icon: "none",
            key: "another_thing",
            value: "simple_value",
            hidden: true,
            action: "none"
          });
          step.displayEvents.push({
            icon: "dot-failure",
            key: "expect_column_values_to_match_regex",
            summary: "show output",
            action: "show-in-modal",
            value: `{
  'success': False,
  'result': {
    'element_count': 36,
    'unexpected_count': 6,
    'unexpected_percent': 0.16666666666666666,
    'unexpected_percent_nonmissing': 0.16666666666666666,
    'missing_count': 0,
    'missing_percent': 0.0,
    'partial_unexpected_counts': [{'value': 'A', 'count': 1}, {'value': 'E', 'count': 5}],
    'partial_unexpected_index_list': [0, 10, 11, 12, 13, 14],
    'partial_unexpected_list': ['A', 'E', 'E', 'E', 'E', 'E']
  }
}`,
            hidden: true
          });

```

**Comment by bengotow** [comment]
One more pass, allowing for each DisplayEvent to specify one or more items which are displayed beneath it in a more-or-less key-value format. (The `text` and `actionText` are both optional, so you can create either a list of items like the first and second example below, or a table of items like the third one)

image



```
export interface IStepDisplayEvent {
  icon:
    | "dot-success"
    | "dot-failure"
    | "dot-pending"
    | "file"
    | "link"
    | "none";
  text: string;
  items: {
    text: string; // shown in gray on the left
    action: "open-in-tab" | "copy" | "show-in-modal" | "none";
    actionText: string; // shown after `text`, optionally with a click action
    actionValue: string; // value passed to the click action
  }[];
}
```

```
step.displayEvents.push({
  icon: "link",
  text: "Materialization",
  items: [
    {
      text: (log.materialization.path || "").split("/").pop()!,
      actionText: "[Copy Path]",
      action: "copy",
      actionValue: log.materialization.path || ""
    }
  ]
});
step.displayEvents.push({
  icon: "dot-success",
  text: "expect_plan_upload_success",
  items: [
    {
      text: "",
      action: "open-in-tab",
      actionText: "https://www.nytimes.com/2019/05/08/...",
      actionValue:
        "https://www.nytimes.com/2019/05/08/us/politics/donald-trump-jr-senate-subpoena.html?action=click&module=Top%20Stories&pgtype=Homepage"
    }
  ]
});
step.displayEvents.push({
  icon: "dot-failure",
  text: "expect_column_values_to_match_regex",
  items: [
    {
      text: "regex:",
      action: "none",
      actionText: "/link?[1-9A-Z]{6}/g",
      actionValue: ""
    },
    {
      text: "output:",
      action: "show-in-modal",
      actionText: "[View]",
      actionValue: `{
'success': False,
'result': {
'element_count': 36,
'unexpected_count': 6,
'unexpected_percent': 0.16666666666666666,
'unexpected_percent_nonmissing': 0.16666666666666666,
'missing_count': 0,
'missing_percent': 0.0,
'partial_unexpected_counts': [{'value': 'A', 'count': 1}, {'value': 'E', 'count': 5}],
'partial_unexpected_index_list': [0, 10, 11, 12, 13, 14],
'partial_unexpected_list': ['A', 'E', 'E', 'E', 'E', 'E']
}
}`
    }
  ]
});
```

**Comment by schrockn** [comment]
I created this pipeline to exercise this UI a bit. I think it might be useful to kick test this a bit. https://github.com/dagster-io/dagster/pull/1339

File: js_modules/dagit/src/DisplayEvent.tsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    version="1.1"
+    style={{
+      flexShrink: 0,
+      alignSelf: "center"
+    }}
+  >
+    <g>
+      <path
+        fill="currentColor"
+        d="M568.2,644.1c-54.4,0-108.7-20.7-150.1-62.1c-11.2-11.2-11.2-29.5,0-40.8c11.2-11.2,29.5-11.2,40.8,0c60.3,60.3,158.4,60.3,218.7,0l209.6-209.6c60.3-60.3,60.3-158.4,0-218.7c-60.3-60.3-158.4-60.3-218.7,0L491.6,289.7c-11.2,11.2-29.5,11.2-40.8,0c-11.2-11.2-11.2-29.5,0-40.8L627.7,72.1c82.8-82.8,217.5-82.8,300.2,0c82.7,82.8,82.8,217.5,0,300.2L718.3,582C676.9,623.4,622.5,644.1,568.2,644.1L568.2,644.1L568.2,644.1z M222.2,990c-54.4,0-108.7-20.7-150.1-62.1c-82.8-82.8-82.8-217.5,0-300.2L281.7,418c82.8-82.8,217.5-82.8,300.2,0c11.2,11.2,11.2,29.5,0,40.8c-11.2,11.2-29.5,11.2-40.8,0c-60.3-60.3-158.4-60.3-218.7,0L112.9,668.4c-60.3,60.3-60.3,158.4,0,218.7c60.3,60.3,158.4,60.3,218.7,0l176.9-176.9c11.2-11.2,29.5-11.2,40.8,0c11.2,11.2,11.2,29.5,0,40.8L372.3,927.9C330.9,969.3,276.6,990,222.2,990L222.2,990L222.2,990z"
```
Comment (alangenfeld) [inline js_modules/dagit/src/DisplayEvent.tsx]: This is intense to have in code inline, I assume you did not manually write this. Not sure the right way to handle this but it is jarring.

Comment (bengotow) [reply js_modules/dagit/src/DisplayEvent.tsx]: Yeah that's a good point - I'll pull these out into actual .svg files. I feel like I had a good reason for inlining them at some point but I can't recall what it was.

**Comment by bengotow** [comment]
Rebased on master and updated the SVGs to be external to the source file.

**Pull request merged**